File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
      int l = 0;
      int r = heights.length-1; 
      int j = 0;
      while(l < r)
      {
        j = Math.max(j, ((Math.min(heights[l], heights[r]))*Math.abs(r-l)));
        if(heights[l] > heights[r])
        {
            r--;
        }
        else
        {
            l++;
        }
      }
      return j;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l = 1 
        r = max(piles)
        res = r 


        while l <= r:
            mid = (l + r) // 2

            totalCount = 0 
            for j in piles:
                totalCount += math.ceil(j/mid)
            if totalCount <= h:
                res = mid 
                r = mid - 1 
            else:
                l = mid + 1
        return res

        

        
            

        


       
       



        
            

        